### PR TITLE
Feature/docs2

### PR DESCRIPTION
**Pull request by oliverturner**
VERY speculative take on docs: have thinned the main entry point in the root README down to reduce visual clutter and extracted feature removal instructions into dedicated linked files.

Looking for feedback.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -24,6 +24,22 @@
 
 ## Overview
 
+### Quickstart
+
+1. First off, let's kick the tyres and look at the sample _Repospective_ app that
```
Comment (mxstbr) [inline docs/README.md]: Repospective :+1:

File: docs/README.md [hunk 2]
```diff
@@ -24,6 +24,22 @@
...
+  comes bundled with this project to demo some of its best features:
+
+    ```Shell
+    npm install && npm start
```
Comment (mxstbr) [inline docs/README.md]: No need to `npm install` here, `npm run setup` takes care of that!

File: docs/README.md [hunk 3]
```diff
@@ -24,6 +24,22 @@
...
+    ```
+
+  And go to [localhost:3000](http://localhost:3000) to see it in action.
+
+  - Add a Github username to see Redux and Redux Sagas in action
+  - Click the Features button to see React Router in action
+  - Edit the file at `/app`
```
Comment (mxstbr) [inline docs/README.md]: This line is confusing, it seems as if one is prompted to go to `localhost:3000/app`, which isn't actually anything

File: docs/general/server-configs.md [hunk 4]
```diff
@@ -4,9 +4,14 @@
...
-1. Redirect all traffic to HTTPS because ServiceWorker only works for encrypted traffic.
+1. Redirect all traffic to HTTPS because ServiceWorker only works for encrypted
+   traffic.
+1. Rewrite all pages (e.g. `yourdomain.com/subpage`) to `yourdomain.com/index.html`
+   to let `react-router` take care of presenting the correct page.
 
-2. Rewrite all pages (e.g. `yourdomain.com/subpage`) to `yourdomain.com/index.html` to let `react-router` take care of presenting the correct page.
+> Note: For performance reasons you should probably adapt it to run as a static
+  `.conf` file (typically under `/etc/apache2/sites-enabled` or similar) so that
+  your server doesn't have to apply its rules dynamically per request)
```
Comment (mxstbr) [inline docs/general/server-configs.md]: Good point :+1:

File: docs/js/redux-saga.md [hunk 5]
```diff
@@ -1,21 +1,36 @@
...
+  click and then update widget2 by dispaching a new event that widget2 is aware of.
+>
+> This adds a level of indirection that is unnecessary for simple apps, but make
+  it more easy to scale complex applications. You can now publish widget1 and
+  widget2 to different npm repositories so that they never have to know about
+  each others, without having them to share a global registry of actions. The 2
+  widgets are now bounded contexts that can live separately. They do not need
+  each others to be consistent and can be reused in other apps as well. **The saga
+  is the coupling point between the two widgets that coordinate them in a
+  meaningful way for your business.**
```
Comment (mxstbr) [inline docs/js/redux-saga.md]: Very good highlighting and copy change :+1:

File: docs/js/reselect.md [hunk 6]
```diff
@@ -5,8 +5,8 @@ tree. This means repeated changes and calculations are fast and efficient,
 providing us with a performance boost over standard `mapStateToProps`
 implementations.
 
-The [official documentation](Lee Bannard
-https://github.com/reactjs/reselect) offers a good starting point!
+The [official documentation](Lee Bannardhttps://github.com/reactjs/reselect)
```
Comment (mxstbr) [inline docs/js/reselect.md]: That link is completely broken I now notice, should fix it while we're at it!

File: docs/testing/component-testing.md [hunk 7]
```diff
@@ -55,30 +55,27 @@ class HomePage extends React.Component {
...
-<button>
-  <CheckmarkIcon />
-  Click Me!          <!-- { props.children } -->
-</button>
-<!-- </Button> -->
+```html
+<button>              <!-- <Button>             -->
+  <CheckmarkIcon />   <!--   NOT RENDERED!      -->
+  Click Me!           <!--   { props.children } -->
+</button>             <!-- </Button>            -->
```
Comment (mxstbr) [inline docs/testing/component-testing.md]: Loving the comments here!

**Comment by mxstbr** [comment]
Huge :+1: all around, this is so much better!! Thanks so much! Three tiny things, but this is absolutely amazing, A+!

File: docs/README.md (hunk 3 above)
Comment (oliverturner) [inline docs/README.md]: Yeah... forgot that this bit was only half-baked :/

(Goes for the previous comment too...)

File: docs/README.md [hunk 8]
```diff
@@ -26,23 +26,36 @@
...
-  - Edit the file at `/app`
+    - Add a Github username to see Redux and Redux Sagas in action: effortless
+      async state updates and side effects are now yours :)
+    - Edit the file at `./app/containers/HomePage/index.js` so that the text of
+      the `<Button>` component reads "Features!!!"... Hot Module Reloading gives
+      you a feedback loop with your UI so smooth it's almost conversational!
+    - Click your (newly emphatic) Features button to see React Router in action...
+      Now you can share a direct link to that content privately over your LAN or
+      globally addressable to any device, anywhere. Not bad for a locally-running
+      Single Page App.
```
Comment (mxstbr) [inline docs/README.md]: Nice, liking this! :+1:

**Comment by mxstbr** [comment]
Awesome, thanks so much Oliver!

**Pull request merged**